xbmc/pvr/settings/PVRStringSettingValues.cpp

```diff
@@ -0,0 +1,51 @@
+/*
+ *  Copyright (C) 2024 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "PVRStringSettingValues.h"
+
+#include "addons/kodi-dev-kit/include/kodi/c-api/addon-instance/pvr/pvr_defines.h"
+#include "guilib/LocalizeStrings.h"
+#include "utils/StringUtils.h"
+
+#include <string>
+
+namespace PVR
+{
+CPVRStringSettingValues::CPVRStringSettingValues(struct PVR_ATTRIBUTE_STRING_VALUE* values,
+                                                 unsigned int valuesSize,
+                                                 const std::string& defaultValue,
+                                                 int defaultDescriptionResourceId /* = 0 */)
+  : m_defaultValue(defaultValue)
+{
+  if (values && valuesSize > 0)
+  {
+    m_values.reserve(valuesSize);
+    for (unsigned int i = 0; i < valuesSize; ++i)
+    {
+      const std::string value{values[i].strValue ? values[i].strValue : ""};
+      const char* desc{values[i].strDescription};
+      std::string strDescr{desc ? desc : ""};
```

Comment:
`strDesc` would be a more commonplace abbreviation. The current var name made me look twice ;)

Reply:
O yes, seems more like a typo + autocomplete afterwards to me. Will change.